lib/concentrate/parser/gtfs_realtime_enhanced.ex

```diff
@@ -152,11 +152,22 @@ defmodule Concentrate.Parser.GTFSRealtimeEnhanced do
     Map.get(vp, "occupancy_status")
   end
 
-  defp decode_state_of_charge(vehicle) do
-    %{
-      "value" => Map.get(vehicle, "state_of_charge_percentage"),
-      "time" => Map.get(vehicle, "state_of_charge_timestamp")
-    }
+  defp decode_state_of_charge(%{
+         "state_of_charge_percentage" => nil,
+         "state_of_charge_timestamp" => nil
+       }),
+       do: nil
+
+  defp decode_state_of_charge(vp) do
+    value = Map.get(vp, "state_of_charge_percentage")
+    time = Map.get(vp, "state_of_charge_timestamp")
+
+    if value && time do
+      %{
+        value: Map.get(vp, "state_of_charge_percentage"),
+        time: Map.get(vp, "state_of_charge_timestamp")
+      }
+    end
```

Comment:
suggestion: I think this might be backwards? I think you'd want the default case to be where both values are present, and the fallback to be `nil`:
```elixir
defp decode_state_of_charge(
  %{
    "state_of_charge_percentage" => pct, 
    "state_of_charge_timestamp" => unix
  }) when is_integer(pct) and is_integer(unix) do
  %{
    value: pct
    time: unix
  }
end
defp decode_state_of_charge(_vp) do
  nil
end
```